server/cluster/cluster.go

```diff
@@ -490,11 +483,30 @@ func (c *RaftCluster) SetStorage(s *core.Storage) {
 	c.storage = s
 }
 
+// GetBasicCluster returns the basic cluster.
+// There is no need a lock since it won't changed.
+func (c *RaftCluster) GetBasicCluster() *core.BasicCluster {
+	return c.core
+}
+
 // GetOpts returns cluster's configuration.
+// There is no need a lock since it won't changed.
 func (c *RaftCluster) GetOpts() *config.PersistOptions {
 	return c.opt
 }
 
+// GetAllocator returns cluster's id allocator.
+// There is no need a lock since it won't changed.
+func (c *RaftCluster) GetAllocator() id.Allocator {
+	return c.id
+}
+
+// GetRuleManager returns the rule manager reference.
+// There is no need a lock since it won't changed.
+func (c *RaftCluster) GetRuleManager() *placement.RuleManager {
+	return c.ruleManager
+}
+
```

Comment:
ditto

Reply:
This function will not be called before the cluster is initialized.